### ISSUE TITLE
Prometheus datasource's Test doesn't take into account URL changes.

### ISSUE THREAD

**Issue by colega** [grafana/grafana#57328]
**What happened**

I went to a datasource's settings page, changed the URL to one that doesn't work, clicked on Test, and it still reported to be OK.

https://user-images.githubusercontent.com/1511481/196892267-4faeb6d0-a47e-4874-80b2-6a3a1bd83440.mp4

**What you expected to happen**:

I expected it to fail.

**How to reproduce it (as minimally and precisely as possible)**:

- Go to a working Prometheus datasource settings
- Change the URL to something that you know doesn't work
- Click on Test

**Anything else we need to know?**:

I tried the same on a non-working datasource that had wrong credentials, changing the URL that would accept those credentials, and it kept failing (although I expected it to succeed). It seems to just ignore the endpoint changes.

**Environment**:
- Grafana version: 9.1.5 (df015...)
- Data source type & version: Prometheus
- OS Grafana is installed on: Linux
- User OS & Browser: Chrome on Linux
- Grafana plugins:
- Others:

**Comment by gtk-grafana** [comment]
@colega This seems to be a provisioned datasource, so you can't change the settings in the UI, you'll need to update the provisioning file if you want to update configuration.

<img width="1146" alt="image" src="https://user-images.githubusercontent.com/109082771/196947213-119f8dba-6c8b-476d-8baa-f59d5e60c89b.png">

**Comment by colega** [comment]
@gtk-grafana if that is the case, why fields are editable? Wouldn't it make sense to make them disabled?

I never wanted to "save" the data source, what I wanted was to see if it would still work if I changed the url to a different one: I changed it and clicked on the "Test" button, the UI told me that it was OK, so I committed the change and the newly provisioned data source failed. 

There are editable fields on that page and there is a test button, as a user I expect to be able to interact with them without having to know the details of how provisioned datasources work (note again, that I never tried to save it).
